src/parser.ts

```diff
@@ -54,321 +55,425 @@ const PLUS = token('+');
 const QUESTION_MARK = token('?');
 const SUBTRACT_MARKER = token('-[');
 
-function asSetOfCodepoints(chars: string): Set<Codepoint> {
-	return new Set(chars.split('').map(c => asCodepoint(c)));
-}
-
-function codepoint(input: string, offset: number): ParseResult<Codepoint> {
-	const codepoint = input.codePointAt(offset);
-	if (codepoint === undefined) {
-		return error(offset, ['any character']);
+export function generateParser(options: { language: string }): (input: string) => RegExp {
```

Comment:
Let's check whether this does not negatively affect the bundle size too much.

Reply:
For the .mjs: Old: 20.318 Bytes, new:  21.055 Bytes
For the .js: Old: 20.570 Bytes, new: 21.304 Bytes

So it's not too bad... Let's re-check when the rest of the features land!